model/src/main/kotlin/utils/DependencyGraphBuilder.kt

```diff
@@ -331,9 +331,19 @@ class DependencyGraphBuilder<D>(
         val dependencies = dependencyHandler.dependenciesFor(dependency)
         if (ref.dependencies.size != dependencies.size) return false
 
-        val dependencies1 = ref.dependencies.map { dependencyIds[it.pkg] }
+        val dependencies1 = ref.dependencies.mapTo(mutableSetOf()) { dependencyIds[it.pkg] }
         val dependencies2 = dependencies.associateBy { dependencyHandler.identifierFor(it) }
-        if (!dependencies2.keys.containsAll(dependencies1)) return false
```

Comment:
BTW, @oheger-bosch do you recall why the original code uses an "uni-directional" `containsAll()` here? In other words, why did we continue here if `dependencies2` was a superset of `dependencies1`, instead of also returning `false` early?

Reply:
I think this is because the two sets only contain identifier IDs, but it has to be checked whether the structure of the dependency sub trees is identical.

Reply:
Hmm, ok, but that does not really answer my question why we can't / shouldn't do

    if (dependencies1 != dependencies2.keys) return false

because if those sets of IDs are not equal, also their sub-trees cannot be equal.

Reply:
Let's anyway extract this a bit unrelated change: https://github.com/oss-review-toolkit/ort/pull/9940